ros2_batch_job/__main__.py

```diff
@@ -389,12 +392,14 @@ def run(args, build_function, blacklisted_package_names=None):
         # in.
         if not args.src_mounted:
             print('# BEGIN SUBSECTION: import repositories')
-            # Get the repositories
-            job.run(['curl', '-sk', args.repo_file_url, '-o', 'ros2.repos'])
-            # Show the contents
-            log("@{bf}==>@| Contents of `ros2.repos`:")
-            with open('ros2.repos', 'r') as f:
-                print(f.read())
+            repos_file_urls = [args.repo_file_url]
+            if args.supplemental_repo_file_url is not None:
+                repos_file_urls.append(args.supplemental_repo_file_url)
+            repos_filenames = []
+            for index, repos_file_url in enumerate(repos_file_urls):
+                repos_filename = '%02d-ros2.repos' % index
```

Comment:
Why not keep the filename? E.g. use the filename from the provided URL and prefix a number (to ensure uniqueness and communicate the order).